tasty/src/dotty/tools/tasty/TastyHeaderUnpickler.scala

```diff
@@ -107,11 +107,20 @@ object TastyHeaderUnpickler {
     }
     val minorVersion = showMinorVersion(MinorVersion, ExperimentalVersion)
     val fileMinorVersion = showMinorVersion(fileMinor, fileExperimental)
-    s"""TASTy signature has wrong version.
-      | expected: {majorVersion: $MajorVersion, minorVersion: $minorVersion}
-      | found   : {majorVersion: $fileMajor, minorVersion: $fileMinorVersion}
-      |
-      |""".stripMargin
+
+    if (MajorVersion == 28 && fileMajor == 28)
+      s"""Found a file on classpath, that was compiled with a newer Scala version than the one currently used.
+         |  Found: 3.$fileMinorVersion
+         |  Supported: 3.$minorVersion or older
+         |
+         | Try updating your Scala version to 3.$fileMinorVersion or change the classpath.""".stripMargin
```

Comment:
```suggestion
         | To fix this issue, try updating your Scala version to 3.$fileMinorVersion.""".stripMargin
```
Since most of the time this is going to happen to a user when they are just adding deps or something, do you think it's necessary to include `or change the classpath`? I feel like for most user they'll have no idea what that actually means, and since this is primarily to help the common programmer, maybe it's better to leave it out? wdyt?